addons/sale_pdf_quote_builder/models/res_company.py

```diff
@@ -10,3 +14,16 @@ class ResCompany(models.Model):
     sale_header_name = fields.Char()
     sale_footer = fields.Binary(string="Footer pages")
     sale_footer_name = fields.Char()
+
+
+    @api.constrains('sale_header')
+    def _ensure_header_not_encrypted(self):
+        for company in self:
+            if company.sale_header:
+                utils._ensure_document_not_encrypted(base64.b64decode(company.sale_header))
+
+    @api.constrains('sale_footer')
+    def _ensure_footer_not_encrypted(self):
+        for company in self:
+            if company.sale_footer:
+                utils._ensure_document_not_encrypted(base64.b64decode(company.sale_footer))
```

Comment:
Couldn't you group them in a single method constraining both of them at the same time? Maybe worse in performance but you wouldn't have two exact same methods.

Reply:
I thought about it too, but IMHO, made more sense to keep them separate as even if it's the same check, these pdf are not related whatsoever. I also kept the same logic as compute: "one field per compute". Don't run the check on the header when you uploaded only a footer, for instance. I could change if needed, though. I did put an utils to avoid repetition though

Reply:
Yes, open question. My thought was more: what if we add another pdf? I don't know, I raised the question, I'll leave you the answer :grin: